src/app/components/projects/pages/details/details.component.ts

```diff
@@ -51,28 +56,109 @@ const projectKey = "project";
         </div>
       </div>
 
-      <p class="lead" *ngIf="!hasSites && !hasRegions">
-        No additional data to display here, try adding sites or regions to the
-        project
+      <h2>
+        Sites
+        <small class="text-muted">
+          <!-- TODO This calculation is inaccurate -->
+          found {{ collectionSizes.sites + collectionSizes.regions }} sites
+        </small>
+      </h2>
+
+      <baw-debounce-input
+        label="Filter"
+        placeholder="Filter Sites"
+        [default]="filter"
+        (filter)="onFilter($event)"
+      ></baw-debounce-input>
+
+      <baw-loading [display]="loading"></baw-loading>
+
+      <p *ngIf="!hasSites && !hasRegions && !loading" class="lead">
+        No additional data to display here, try adding sites to the project
       </p>
 
-      <app-site-cards *ngIf="hasSites" [project]="project"></app-site-cards>
-      <app-region-cards
-        *ngIf="hasRegions"
-        [project]="project"
-      ></app-region-cards>
+      <div id="model-grid">
+        <!-- Google Maps -->
+        <div *ngIf="hasSites || hasRegions" class="item map">
+          <app-site-map [project]="project"></app-site-map>
+        </div>
+
+        <!-- Regions -->
+        <div *ngFor="let region of regions" class="item">
+          <app-site-card [project]="project" [region]="region"></app-site-card>
+        </div>
+
+        <!-- Sites -->
+        <div *ngFor="let site of sites" class="item">
+          <app-site-card [project]="project" [site]="site"></app-site-card>
+        </div>
+      </div>
+
+      <ngb-pagination
+        *ngIf="displayPagination"
+        aria-label="Pagination Buttons"
+        class="mt-2 d-flex justify-content-end"
+        [collectionSize]="collectionSize"
+        [(page)]="page"
+      ></ngb-pagination>
     </ng-container>
   `,
   styleUrls: ["./details.component.scss"],
 })
-class DetailsComponent extends PageComponent implements OnInit {
+class DetailsComponent extends PaginationTemplate<any> implements OnInit {
+  public collectionSize = 0;
+  public collectionSizes = { sites: 0, regions: 0 };
   public defaultDescription = "<i>No description found</i>";
-  public project: Project;
   public hasRegions: boolean;
   public hasSites: boolean;
+  public project: Project;
+  public regions: List<Region>;
+  public sites: List<Site>;
+  /**
+   * Api requests are independent, use this to track when both complete
+   */
+  private apiReturnCount = 0;
```

Comment:
requests may be independent, but a hook after the [merge](https://github.com/QutEcoacoustics/workbench-client/pull/511/files#diff-3fc28901b5267a7f8ed292901d944f39R180) could set this to true/false

Reply:
I'm not really sure how to do this, I can't use `complete` because the parent observable never completes. Alternatively, i could use `last` to read the last value of the merge observable, however that doesn't seem to be possible within the constraints of the abstract parent class